tests/Controller/FeedControllerTest.php

```diff
@@ -38,18 +41,63 @@ public function testSubmitAndCreatePost(): void
         $this->assertTrue($newPost->getPostText() === 'testing post');
     }
 
-    public function testItCanEditAnExistingPost(): void
+    public function testSubmitAndCreateComment(): void
     {
         $testUser = $this->dogRepo->findOneByUsername('testUser');
         $this->client->loginUser($testUser);
-
-        // We will need to find a new form here to do the edit
-        // Posts exist for the testUser already and their content is set. Just need to update it
-        // Should assert that the db was updated
+        $testUserPost = $this->postRepo->findBy(['post_text' => 'Woof woofingtons'])[0];
+        $url = "/post/{$testUserPost->getId()}/add-comment";
+        $crawler = $this->client->request('GET', $url);
+        $form = $crawler->selectButton('Save')->form();
+        $form["comment_form[comment_text]"] = 'testing comment';
+        $crawler = $this->client->submit($form);
+        
+        $newComment = $this->commentRepo->findBy(['comment_text' => 'testing comment'])[0];
+        $this->assertTrue($newComment->getCommentText() === 'testing comment');
     }
 
-    public function testItCanDeleteAPost(): void
+    public function testAddDogToPack(): void
     {
-        // Should be able to log in as a user and delete a post by id
+        $testUser = $this->dogRepo->findOneByUsername('testUser');
+        $this->client->loginUser($testUser);
+
+        $testUserPack = $this->dogRepo->getDogsPack($testUser);
+
+        $url = "/dog/{$testUser->getId()}/add-friend";
+        $this->client->request('GET', $url);
+        
+        $newTestUserPack = $this->dogRepo->getDogsPack($testUser);
+        $this->assertTrue((count($testUserPack) + 1) === count($newTestUserPack));
```

Comment:
Love this test. Seriously well done 🔥 

The only thing I'd say to improve this is to not use count as your assertion as it doesn't confirm actual data. This is a tiny thing though and IMO may be slightly preferable as you aren't coupling your tests to your fixture.

*if* you want to change this, take a look at the methods on the ArrayCollection https://www.doctrine-project.org/projects/doctrine-collections/en/1.6/index.html

Reply:
I found a alternative too, assertContains & assertNotContains. https://phpunit.readthedocs.io/en/9.5/assertions.html#assertcontains.

Reply:
I also needed to look at this within the removal of dogs from a pack so, that resource you shared was golden. :)